### PR TITLE
test: make test test_destroy_missing more stable

### LINKED ISSUE #15615: unstable test test_destroy_missing
### What operating system and CPU are you using?

```
[2023-09-14T21:37:28.854Z] --- TRY 3 STDERR:        raftstore-v2 worker::tablet::tests::test_destroy_missing ---

[2023-09-14T21:37:28.854Z] thread 'worker::tablet::tests::test_destroy_missing' panicked at 'assertion failed: !path.exists()', components/raftstore-v2/src/worker/tablet.rs:817:9

[2023-09-14T21:37:28.854Z] stack backtrace:

[2023-09-14T21:37:28.854Z]    0: rust_begin_unwind

[2023-09-14T21:37:28.854Z]              at /rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/std/src/panicking.rs:575:5

[2023-09-14T21:37:28.854Z]    1: core::panicking::panic_fmt

[2023-09-14T21:37:28.854Z]              at /rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/core/src/panicking.rs:65:14

[2023-09-14T21:37:28.855Z]    2: core::panicking::panic

[2023-09-14T21:37:28.855Z]              at /rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/core/src/panicking.rs:114:5

[2023-09-14T21:37:28.855Z]    3: raftstore_v2::worker::tablet::tests::test_destroy_missing

[2023-09-14T21:37:28.855Z]    4: raftstore_v2::worker::tablet::tests::test_destroy_missing::{{closure}}

[2023-09-14T21:37:28.855Z]    5: core::ops::function::FnOnce::call_once

[2023-09-14T21:37:28.855Z]    6: core::ops::function::FnOnce::call_once

[2023-09-14T21:37:28.855Z]              at /rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/core/src/ops/function.rs:513:5

[2023-09-14T21:37:28.855Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### PR DESCRIPTION
**Pull request by glorv**
### What is changed and how it works?

Issue Number: Close #15615

What's Changed:

```
Add a loop to wait async pause_background_work task finish before trying to clean up tablet.
The `pause_background_work` task can block tablet's close because it also hold a valid arc reference.
```

### Check List 

Tests 

- Unit test

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore-v2/src/worker/tablet.rs [hunk 1]
```diff
@@ -822,6 +829,13 @@ mod tests {
         runner.run(Task::destroy(r_1, 100));
         assert!(path.exists());
         registry.remove(r_1);
+        // waiting for async `pause_background_work` to be finished,
+        // this task can block tablet's destroy.
+        for _i in 0..100 {
+            if runner.get_running_task_count() > 0 {
+                std::thread::sleep(Duration::from_millis(5));
+            }
```
[the remark this answers is not in the document]
Comment (glorv) [reply components/raftstore-v2/src/worker/tablet.rs:837]: 😅

**Review by Connor1996 — APPROVED** [review]
LGTM

**Comment by glorv** [comment]
@overvenus @SpadeA-Tang  PTAL

**Review by SpadeA-Tang — APPROVED** [review]

**Pull request merged**